### PR TITLE
Feat/theming

### LINKED ISSUE #145: Expose theme-ing apis for use-ck UI
expose tailwind and show howto in docs

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
This is a big one that needs some discussing. We may want to take a different approach. I tried to stay backwards compatible with the dark and light theme. But this can easily be dropped in favour of providing themes during app's setup (or via `updateTheme`).

Fix #145

### CODE REVIEW

File: packages/example/components/theme-buttons.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+function ThemeButton({
+  theme,
+  onClick,
+}: Props & {
+  theme: Theme;
+}) {
+  return (
+    <div
```
Comment (aaronmgdr) [inline packages/example/components/theme-buttons.tsx:14]: Please Make this a `<button>` for accessibility

File: packages/example/components/theme-buttons.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      <div className="flex flex-row gap-x-2">
+        <div
+          className="w-3 h-full rounded"
+          style={{ background: theme.primary }}
+        />
+        <div
+          className="w-3 h-full rounded"
+          style={{ background: theme.secondary }}
+        />
+      </div>
```
Comment (aaronmgdr) [inline packages/example/components/theme-buttons.tsx:46]: This is pretty cool

File: packages/react-celo/src/components/button.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+export default function Button({
+  as = 'button',
+  className,
+  style,
+  ...props
+}: Props) {
+  const theme = useTheme();
+
+  const Component = as;
+
```
Comment (aaronmgdr) [inline packages/react-celo/src/components/button.tsx:33]: Ive never seen the `as = 'element'`, `Component = as` before.

File: packages/react-celo/src/screens/metamask.tsx [hunk 4]
```diff
@@ -31,47 +32,39 @@ const styles = cls({
...
+  const theme = useTheme();
   const isMetaMask = isEthereumFromMetamask();
   const { error } = useInjectedConnector(onSubmit, isMetaMask);
 
   let content: React.ReactElement;
 
   if (isMobile) {
     content = (
-      <a
+      <Button
```
Comment (aaronmgdr) [inline packages/react-celo/src/screens/metamask.tsx:52]: missing `as =  'a'`

File: packages/react-celo/src/components/button.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+type extendType = anchorType & buttonType;
+
+interface Props extends extendType {
+  as?: 'a' | 'button' | React.ComponentType<unknown>;
+  className?: string;
+  style?: Record<string, string | number>;
+}
+
+export default function Button({
+  as = 'button',
```
Comment (aaronmgdr) [inline packages/react-celo/src/components/button.tsx:25]: id like to make this case that this should be required so we dont accidentally make buttons when anchor tags are the semantically correct thing to do

File: packages/react-celo/src/utils/useTheme.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      subtree: false,
+      childList: false,
+      attributeFilter: ['class'],
+    });
+  });
+}
+
+export default function useTheme(): Theme {
+  const [{ theme }] = useReactCeloContext();
+  console.log(theme);
```
Comment (aaronmgdr) [inline packages/react-celo/src/utils/useTheme.ts:64]: remove

**Review by aaronmgdr — CHANGES_REQUESTED** [review]
mostly just questions but There was one button that should be an `a`

File: packages/example/pages/index.tsx [hunk 7]
```diff
@@ -39,6 +40,12 @@ const feeTokenMap: FeeTokenMap = {
 function truncateAddress(address: string) {
   return `${address.slice(0, 8)}...${address.slice(36)}`;
 }
+const html =
+  typeof document !== 'undefined' &&
+  (document.getElementsByTagName('html')[0] as HTMLElement);
+function isDark() {
+  return html && html.classList.contains('tw-dark');
+}
```
[the remark this answers is not in the document]
Comment (nicolasbrugneaux) [reply packages/example/pages/index.tsx:48]: maybe? it's in the example app I didn't quite look for optimization there

File: packages/react-celo/src/utils/useTheme.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  useEffect(() => {
+    const cb = (m: Mode) => setMode(m);
+    state.addEventListener(cb);
+    return () => {
+      state.removeEventListener(cb);
+    };
+  }, []);
+
+  return theme || defaultTheme[mode];
+}
```
Comment (nicolasbrugneaux) [inline packages/react-celo/src/utils/useTheme.ts:75]: Note: That hook could be soooo much simpler if we didn't handle the html or body dark or tw-dark classnames.

File: packages/react-celo/__tests__/react-celo-provider.tsx [hunk 9]
```diff
@@ -172,11 +180,44 @@ describe('CeloProvider', () => {
...
           expect(result.current.supportsFeeCurrency).toBe(false);
         });
       });
     });
+
+    it('updates the current theme', () => {
+      const { result, rerender } = renderUseCeloInternal({ theme: null });
+
+      // FIXME Need to determine behavior when network is not in networks
+      expect(result.current.network).toEqual(Mainnet);
```
Comment (aaronmgdr) [inline packages/react-celo/__tests__/react-celo-provider.tsx:194]: I think this line can be remove right? copied from another test?

**Review by aaronmgdr — APPROVED** [review]

**Pull request merged**